### PR TITLE
Switch server

### PR DESCRIPTION
**Pull request by jwnasambu**
## Requirements

- [ ] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [ ] I have tested the changes locally and verified that they work as intended.
- [ ] I have added comments to my code where necessary to help others understand my changes.
- [ ] I have read and agree to the Contributing Guidelines of this project.

## Summary
I switched the test server from local server to UAT server in preparation for 2.13.0-Snapshot testing.

## Screenshots

*None.*

## Related Issue

*None.*

## Other

*None.*

### CODE REVIEW

**Comment by kdaud** [comment]
Thanks @jwnasambu for putting this up. 

I have skimmed through the logs and most probably the issue could be from one of the properties residing in the test-local-property files. Try to get rid of the `db.host` property!

**Comment by jwnasambu** [comment]
@kdaud kindly I have removed `db.host property!` but still flopping

**Comment by kdaud** [comment]
Change the` includes.csrftoken` property value to `false`.

**Comment by jwnasambu** [comment]
@kdaud some tests are still failing

**Comment by kdaud** [comment]
@jwnasambu we need to have the test run when the includes.csrftoken=true. Revert the recent commit then we figure out the issue.

**Comment by kdaud** [comment]
@jwnasambu could you have an idea of why the automated tests are failing with uat-server for RefApp 2.13.0-SNAPSHOT?

**Comment by jwnasambu** [comment]
@kdaud looking at the docker yml file on this link https://github.com/openmrs/openmrs-contrib-qaframework/blob/3312958b4ac4539d56f2828aea87a944041f77e1/docker-compose.yml#LL18C2-L18C59 it seems the configuration was made for QA server  as opposed to the UAT. That is my silly thinking

**Comment by kdaud** [comment]
> it seems the configuration was made for QA server as opposed to the UAT

Not really, the idea is that the local instance we are using to run the tests in github action should use an updated docker image of qa which contains the latest changes. So by the change of this makes the config ignored because we are setting the server not to be local host. Did you notice that some tests were passing when you set the `includes.csrftoken` to false?

**Comment by kdaud** [comment]
Do the automated tests pass locally when you set them to run on the uat server?

**Comment by jwnasambu** [comment]
@kdaud I haven't tried locally because I desired to first ensure the workflows are passing when I trigger  from the github actions. Besides, when I  change `includes.csrftoken` to True the is one workflow which is passing. My assumption is to ensure the `includes.csrftoken` is in its right state irrespective of the build failure and begin debugging from there.

cc @ibacher, @dkayiwa, @sherrif10

**Comment by jwnasambu** [comment]
@kdaud I suppose this is the major error `  Unable to locate element: *[name='OWASP-CSRFTOKEN']` we have to deal with as the build failure is concerned.

**Comment by jwnasambu** [comment]
@kdaud I have changed the`includes.csrftoken`  to false for testing the 2.13 version of the RefApp, as that’s a feature added in 2.6 of the platform.

**Comment by kdaud** [comment]
@mherman22 is the qa-refapp server using the released version of platform 2.6.0?

**Comment by kdaud** [comment]
> My assumption is to ensure the `includes.csrftoken` is in its right state irrespective of the build failures and begin debugging from there. We can't run away from the problem.

I agree!

**Comment by kdaud** [comment]
@mherman22 is the qa-refapp server using the released version of platform 2.6.0?

**Comment by mherman22** [comment]
> @mherman22 is the qa-refapp server running on released version of [platform 2.6.0](https://talk.openmrs.org/t/openmrs-platform-2-6-0-released/38585)?

qa-refapp server is running on platform 2.5.9 as can be seen at https://github.com/openmrs/openmrs-distro-referenceapplication/blob/4a083070d1f1f6543fed74a0a4f33dbfe4a68dd4/pom.xml#L30

**Comment by mherman22** [comment]
@kdaud here is the PR that downgraded it from platform 2.6.0.   --> https://github.com/openmrs/openmrs-distro-referenceapplication/pull/694

**Comment by kdaud** [comment]
> I have changed the`includes.csrftoken` to false for testing the 2.13 version of the RefApp, as that’s a feature added in 2.6 of the platform.

@jwnasambu did you notice some tests are failing the builds? Do these tests fail when you run them locally on the uat instance?

**Comment by jwnasambu** [comment]
@kdaud on the local server the random tests I have tried out are failing. One of the workflows is the PatientVisit and this is the error log https://pastebin.com/U4sPVYEN

**Comment by mherman22** [comment]
> @kdaud on the local server the random tests I have tried out are failing. One of the workflows is the PatientVisit and this is the error log https://pastebin.com/U4sPVYEN

Often time, this is a network issue. Can you set `headless to false` or Change URL to run from a local instance setup using either openmrs SDK(not sure if the SDK has been updated with the version of refapp youre releasing) or setup using the docker image you released.

**Comment by jwnasambu** [comment]
@mherman22 my headless is on `false`. Besides, the server starts on the browser but can't go beyond the login form.

**Comment by mherman22** [comment]
> my headless is on `false`

sorry i meant `true` :smile: . However, for the  tests to run successfully, your network needs to be so good. if you dont, then using a local instance of openmrs is a better option.

**Comment by jwnasambu** [comment]
@mherman22 Kindly help me test on your side if its working as expected please!

**Comment by jwnasambu** [comment]
@mherman22 kindly when I change `headless:false` won't I fail to see the tests running on the browser? Besides, I changed `includes.csrftoken=false` and run this test `refapp2OpenConceptLab` and its passed!

File: qaframework-bdd-tests/docker/docker-compose-refqa.yml [hunk 1]
```diff
@@ -29,7 +29,7 @@ services:
       DB_AUTO_UPDATE: 'false'
       MODULE_WEB_ADMIN: 'false'
     healthcheck:
-      test: ["CMD", "curl", "-f", "http://localhost:8080/openmrs/"]
+      test: ["CMD", "curl", "-f", ""https://uat-refapp.openmrs.org/openmrs/""]
```
Comment (ibacher) [inline qaframework-bdd-tests/docker/docker-compose-refqa.yml:32]:
```suggestion
      test: ["CMD", "curl", "-f", "http://localhost:8080/openmrs/"]
```

**Comment by kdaud** [comment]
Looking through the test failures there is a general issue with the uat server throwing [HTTP 404 Not Found](https://github.com/openmrs/openmrs-contrib-qaframework/actions/runs/4360243358/jobs/7622938420#step:7:6337) which needs to be investigated. @jwnasambu is the docker image of [uat](https://github.com/openmrs/openmrs-contrib-qaframework/pull/478/files#:~:text=image%3A%20openmrs/openmrs%2Dreference%2Dapplication%2Ddistro%3Auat) updated?

@mherman22 have you had an experience of testing out the uat server with the automated tests locally?

**Comment by Piumal1999** [comment]
Hi @jwnasambu ,
Is this Pull Request created just for testing purposes or is it meant to be merged into the main branch? I just wanted to confirm.
Thank you

**Pull request closed without merging**